File: CGI/cgi.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    cgi.py                                             :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: dvandenb <dvandenb@student.42.fr>          +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/02/13 14:46:49 by dvandenb          #+#    #+#              #
#    Updated: 2024/02/13 14:46:55 by dvandenb         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import time
import os


def	getCalcul(calcul):
	calcul = calcul.replace("+", " ")
	calcul = calcul.replace("%2B", "+")
	calcul = calcul.replace("%2F", "/")
	return calcul

def	doCGI():
		tm = time.strftime("%A, %d %B %Y, %H:%M")
		try:
			calcul = os.environ["calcul"]
			calcul = getCalcul(calcul)
			result = eval(calcul)
			calcul += " = <b>"
		except:
			calcul = "<b>Invalid Input"
			result = ""

		html1 = """
		<!DOCTYPE html>
		<html lang="en">
			<head>
				<meta charset="utf-8" />
			</head>
			<body>
				<title>Python CGI</title>
				<style>

					body{
						background-color: #252525;
						color: rgb(148, 145, 145);
						font-size: 15px;
						text-align: center;
					}
					a {
						text-decoration: none;
						color: rgb(168, 165, 165);
					}
				</style>
		"""
		html2 = "\n<h1>Python CGI</h1>\n"
				
		html3 = "</body>\n</html>"
		tm ="<p>" + tm + "</p>"
		if (result == 69):
			result = "<a href=\"https://bit.ly/3UH3TNg\" target=\"_blank\">Secret web page</a>"
		elif (result == 42):
			result = "<a href=\"https://bit.ly/49bGzfa\" target=\"_blank\">Secret web page</a>"
		value = "<p>" + calcul + str(result) + "</b></p>"

		print(html1 + tm + html2 + value + html3)

doCGI()

File: CGI/cgi.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   cgi.cpp                                            :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: dvandenb <dvandenb@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/02/13 14:46:39 by dvandenb          #+#    #+#             */
/*   Updated: 2024/02/20 14:42:25 by dvandenb         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <cstdlib>
#include <string>
#include <ctime>
#include <cstdio>
#include <cstring>
#include <dirent.h>
#include <filesystem>

int main()
{
	const char* val = std::getenv("inputNum");
	const char* folder = std::getenv("uploadFolder");
	const char* path_t = std::getenv("PWD");

	if (!val || std::atoi(val) < 1 || std::atoi(val) > 3 || !folder || !path_t)
		return (EXIT_FAILURE);
	std::string temp1(folder);
	std::string temp2(path_t);
	std::string temp3 =  temp2 + "/" + temp1;
	const char* full = temp3.c_str();
	std::string path(folder);
	std::srand(std::time(nullptr));
	std::cout <<"<!DOCTYPE html>"
					"<html lang=\"en\">"
						"<head>"
							"<meta charset=\"utf-8\" />"
						"</head>"
						"<body>"
							"<title>Python CGI</title>"
							"<style>"
								"body{"
									"background-color: #252525;"
									"color: rgb(148, 145, 145);"
									"font-size: 15px;"
									"text-align: center;"
								"}"
							"</style>"
							"<h1>C++ CGI</h1>"
						"</body>";
	if (std::atoi(val) == std::rand() % 3 + 1)
	{
		DIR *directory = opendir(full);
		struct dirent *entry;
		if (directory != NULL) {
        	while ((entry = readdir(directory)) != NULL) {
            	if (strcmp(entry->d_name, ".") && strcmp(entry->d_name, "..")) {
            	    char file_path[256];
            	    sprintf(file_path, "%s/%s", full, entry->d_name);
            	    if (remove(file_path) != 0)
            	        std::cout << "<p>Error deleting file: " << file_path << "</p>";
            	    else 
            	        std::cout << "<p>Deleted file: " << file_path << "</p>";
            	}
        	}
        closedir(directory);
		} else {
			std::cout << "<p>Error opening directory " << std::string(full) << ".</p>";
			std::cout << "</html>";
			return (EXIT_SUCCESS);
		}
	}
	else
		std::cout << "<p>No files deleted!</p>";
		std::cout << "</html>";
	return (EXIT_SUCCESS);
}